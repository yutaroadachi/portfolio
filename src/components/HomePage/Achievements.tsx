import {
  Accordion,
  AccordionButton,
  AccordionIcon,
  AccordionItem,
  AccordionPanel,
  chakra,
  Link,
  Stack,
} from '@chakra-ui/react'
import React from 'react'

export const Achievements = () => (
  <Stack spacing={4} py={4}>
    <chakra.h2 textStyle="heading" fontSize="2xl" textAlign="center">
      Achievements
    </chakra.h2>
    <Accordion defaultIndex={[0]} allowMultiple>
      <Stack spacing={4}>
        {achievements.map((achievement) => (
          <AccordionItem key={achievement.title}>
            <AccordionButton>
              <chakra.p flex="1" textAlign="left">
                {achievement.title}
              </chakra.p>
              <AccordionIcon />
            </AccordionButton>
            <AccordionPanel>
              <Stack spacing={1}>
                <chakra.p>{achievement.description}</chakra.p>
                {achievement.href && (
                  <Link href={achievement.href} isExternal color="link">
                    {achievement.linkTitle ?? achievement.href}
                  </Link>
                )}
              </Stack>
            </AccordionPanel>
          </AccordionItem>
        ))}
      </Stack>
    </Accordion>
  </Stack>
)

type Achievement = {
  title: string
  description: string
  href?: string
  linkTitle?: string
}

const achievements: Achievement[] = [
  {
    title: 'Next.jsに関する記事がZennのTrendingでTech部門2位を達成',
    description:
      '「Next.jsにおけるSSG/SSR/ISR/CSRの違いと使い分け」の記事がZennのTrendingでTech部門2位を達成。その他にも「Next.js最速セットアップ」の記事が同6位を達成するなど、Next.jsに関する知見あり。',
    href: 'https://zenn.dev/a_da_chi/articles/105dac5573b2f5',
    linkTitle: 'Zennで読む',
  },
  {
    title: '映画『僕とオトウト』公式HP製作',
    description:
      '知人が撮影した映画『僕とオトウト』の公式HPを製作。使用技術はSTUDIO。ディレクション、デザイン、実装、リリース（独自ドメイン対応など）と全ての行程を担当。映画の内容が監督自身が重度の知的障害をもつ弟にカメラを向けたドキュメンタリーであったことから、映画の世界観を壊さないデザインとSTUDIOでできる範囲のアクセシビリティ（画像のalt設定やセマンティックなマークアップなど）を意識。',
  },
]
