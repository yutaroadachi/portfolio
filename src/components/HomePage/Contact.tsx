import { chakra, Link, VStack } from '@chakra-ui/react'
import React from 'react'

export const Contact = () => (
  <VStack spacing={4} py={4}>
    <chakra.h2 textStyle="heading" fontSize="2xl">
      Contact
    </chakra.h2>
    <chakra.p>
      お問い合わせフォームは
      <Link
        href="https://docs.google.com/forms/d/e/1FAIpQLSdSMQFzMIm2oWZ9fk_r1RGIUl_gaf3r6ouHVK_Ysn2FERGxVA/viewform?usp=sf_link"
        isExternal
        color="link"
      >
        こちら
      </Link>
    </chakra.p>
  </VStack>
)
