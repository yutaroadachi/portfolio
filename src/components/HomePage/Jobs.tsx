import {
  Accordion,
  AccordionButton,
  AccordionIcon,
  AccordionItem,
  AccordionPanel,
  Box,
  chakra,
  Link,
  Stack,
} from '@chakra-ui/react'
import React from 'react'

export const Jobs = () => (
  <Box py={6}>
    <chakra.h2 textStyle="h2">Jobs</chakra.h2>
    <Accordion defaultIndex={[0]} allowMultiple>
      <Stack spacing={4}>
        {jobs.map((job) => (
          <AccordionItem key={job.companyName}>
            <AccordionButton>
              <Box flex="1" textAlign="left">
                <chakra.p>{job.companyName}</chakra.p>
                <chakra.p fontSize="xs" fontWeight="normal" color="secondary">
                  {job.period}
                </chakra.p>
              </Box>
              <AccordionIcon />
            </AccordionButton>
            <AccordionPanel>
              <Stack spacing={1}>
                <chakra.p>{job.description}</chakra.p>
                {job.href && (
                  <Link href={job.href} isExternal color="link">
                    会社HPを見る
                  </Link>
                )}
              </Stack>
            </AccordionPanel>
          </AccordionItem>
        ))}
      </Stack>
    </Accordion>
  </Box>
)

type Job = {
  companyName: string
  period: string
  description: string
  href?: string
}

const jobs: Job[] = [
  {
    companyName: '株式会社OneMile Partners',
    period: '2020/9 ~ 現職',
    description:
      '「はたらく世代」のお客様向けに、資産運用のサポートを提供する会社にて、業務支援のための管理画面やお客様向けサービスサイトの開発に従事。TypeScript, Next.js, Ruby on Railsをメインにフロントエンドからバックエンドまで幅広く経験。また、スクラム開発によるチーム開発も経験。',
    href: 'https://onemile.co.jp/',
  },
  {
    companyName: '株式会社みんせつ',
    period: '2019/8 ~ 2020/8',
    description:
      '上場企業のIR部門と機関投資家、証券会社をつなぐプラットフォームを提供する会社にて、メインプロダクトのSaaSの企画・開発に従事。React, Ruby on Railsをメインにフロントエンドからバックエンドまで幅広く経験。',
    href: 'https://msetsu.com/',
  },
  {
    companyName: 'NTTコムウェア株式会社',
    period: '2018/4 ~ 2019/7',
    description:
      'システムインテクグレータの会社にて、某金融機関に常駐して社内稟議などのペーパーレス化システムの保守に従事。保守チーム（5名程度）のリーダーとして顧客折衝やマネジメント、資料作成を経験。また開発面では、テスト設計やリリース作業を自動化するbashスクリプトの製造を経験。',
    href: 'https://www.nttcom.co.jp/',
  },
]
