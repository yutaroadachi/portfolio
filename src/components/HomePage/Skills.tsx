import { Box, Center, chakra, ListItem, UnorderedList } from '@chakra-ui/react'
import React from 'react'

export const Skills = () => (
  <Box py={6}>
    <chakra.h2 textStyle="h2">Skills</chakra.h2>
    <Center>
      <UnorderedList>
        {skills.map((skill) => (
          <ListItem key={skill}>{skill}</ListItem>
        ))}
      </UnorderedList>
    </Center>
  </Box>
)

const skills = [
  'TypeScript',
  'React',
  'Next.js',
  'Chakra UI',
  'Ruby on Rails',
  'Go',
]
