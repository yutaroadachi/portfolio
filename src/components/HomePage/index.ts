export * from './Achievements'
export * from './Biography'
export * from './Contact'
export * from './Jobs'
export * from './RecentPosts'
export * from './Skills'
