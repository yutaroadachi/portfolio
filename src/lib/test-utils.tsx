import { ChakraProvider } from '@chakra-ui/react'
import { render } from '@testing-library/react'
import React from 'react'

const AllTheProviders = ({ children }: { children: JSX.Element }) => {
  return <ChakraProvider>{children}</ChakraProvider>
}

const customRender = (ui: JSX.Element, options?: any) =>
  render(ui, { wrapper: AllTheProviders, ...options })

export * from '@testing-library/react'
export { customRender as render }
