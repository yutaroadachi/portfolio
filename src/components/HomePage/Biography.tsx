import { HStack, Link, Text, VStack } from '@chakra-ui/react'
import React from 'react'
import { Image } from 'src/components/common'
import { GitHubIcon, TwitterIcon, ZennIcon } from 'src/components/icons'

export const Biography = () => (
  <VStack spacing={4} layerStyle="card" p={4}>
    <Image
      src="/assets/home-page/biography.png"
      width={150}
      height={150}
      alt="プロフィール画像"
      borderRadius="50%"
    />
    <Text>Yutaro Adachi / Web Engineer</Text>
    <HStack spacing={8}>
      <Link href="https://github.com/yutaroadachi" isExternal>
        <GitHubIcon boxSize="32px" />
      </Link>
      <Link href="https://zenn.dev/a_da_chi" isExternal>
        <ZennIcon boxSize="32px" />
      </Link>
      <Link href="https://twitter.com/a_da_chi_" isExternal>
        <TwitterIcon boxSize="32px" />
      </Link>
    </HStack>
  </VStack>
)
