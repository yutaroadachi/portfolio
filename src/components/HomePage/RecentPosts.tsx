import { Box, Grid, HStack, Link, Text, VStack } from '@chakra-ui/react'
import React from 'react'
import { ZennIcon } from 'src/components/icons'
import { formatYYYYMMDD } from 'src/lib/date-format'
import { RssFeedItem } from 'src/types/rss'

export const RecentPosts = ({ posts }: { posts: RssFeedItem[] }) => (
  <VStack spacing={4} py={4}>
    <Text as="h2" textStyle="heading" fontSize="2xl">
      Recent Posts
    </Text>
    <Grid
      templateColumns={{ base: 'repeat(1, 1fr)', md: 'repeat(2, 1fr)' }}
      gap={4}
    >
      {posts.length > 0 ? (
        posts
          .slice(0, 6)
          .map((post) => <PostCard key={post.link} post={post} />)
      ) : (
        <Text>failed to get recent posts.</Text>
      )}
    </Grid>
    <Link href="https://zenn.dev/a_da_chi" isExternal>
      Read more
    </Link>
  </VStack>
)

const PostCard = ({ post }: { post: RssFeedItem }) => (
  <Link href={post.link} isExternal layerStyle="card" p={4}>
    <HStack spacing={4} h="full">
      <ZennIcon boxSize="32px" />
      <Box>
        <Text fontWeight="bold">{post.title}</Text>
        <Text fontSize="xs" color="secondary">
          {`Published at ${formatYYYYMMDD(new Date(post.pubDate))}`}
        </Text>
      </Box>
    </HStack>
  </Link>
)
