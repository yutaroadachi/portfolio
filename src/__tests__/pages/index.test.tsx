import React from 'react'
import { render } from 'src/lib/test-utils'
import HomePage from 'src/pages'

test('ホームページ', () => {
  const { container } = render(<HomePage />)
  expect(container).toMatchSnapshot()
})
