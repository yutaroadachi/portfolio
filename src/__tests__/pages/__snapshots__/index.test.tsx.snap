// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ホームページ 1`] = `
<div>
  <div
    class="css-1u8qly9"
  >
    <h2
      class="chakra-heading css-eo8fjx"
    >
      Hello World!
    </h2>
  </div>
</div>
`;
